main.go

```diff
@@ -48,6 +48,10 @@ type config struct {
 	// External URI to /api
 	APIPrefix string `yaml:"api_prefix"`
 
+	// Allowed rageshake app names
+	AllowedAppNames   []string `yaml:"allowed_app_names"`
+	allowedAppNameMap map[string]bool
```

Comment:
I'm a bit unconvinced that we should be putting derived properties in this structure; in my opinion it represents exactly the settings we read from the file, and if we want to preprocess settings, then that should go elsewhere.

My approach to this would be to add `allowedAppNameMap` as a separate property in [`submitServer`](https://github.com/matrix-org/rageshake/blob/e61c48177d4c40114abae9dcc31b277494979eaf/submit.go#L49). To populate it, I would create a new function `NewSubmitServer` which makes a new `submitServer`.

Reply:
Ah yes, having looked at it again, it looks like the rest of the configuration that is generated is done when passing it into the creation of a submitServer struct, it seems OK to just add another option [in there](https://github.com/matrix-org/rageshake/pull/67/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R161) - i don't think we need to make a function for just this part of the problem; other than to factor it out of the main method (which is actually one big NewSubmitServer function).